### ISSUE TITLE
store/redis/rdsstore.c:1082:13: error: passing argument 2 of ‘cmp_read_integer’ from incompatible pointer type [-Werror]

### ISSUE THREAD

**Issue by brunoqc** [slact/nchan#232]
nchan 1.0.0
nginx-1.11.3
Ubuntu 14.04.5 LTS (32-bit)

```
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -DNDK_SET_VAR  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/http/modules/perl -I /home/bbigras/nginx/nginx-1.11.3/debian/modules/ngx_devel_kit/objs -I objs/addon/ndk -I /home/bbigras/nginx/nginx-1.11.3/debian/modules/lua-nginx-module/src/api -I debian/extra/njs-0.1.0/nginx/../nxt -I debian/extra/njs-0.1.0/nginx/../njs -I debian/extra/njs-0.1.0/nginx/../nxt -I debian/extra/njs-0.1.0/nginx/../njs -I /opt/openssl-1.0.2h/.openssl/include -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 -I /home/bbigras/nginx/nginx-1.11.3/debian/modules/ngx_devel_kit/src -I /home/bbigras/nginx/nginx-1.11.3/debian/modules/ngx_devel_kit/src -I /home/bbigras/nginx/nginx-1.11.3/debian/modules/ngx_devel_kit/objs -I objs/addon/ndk -I /usr/include/lua5.1 -I /home/bbigras/nginx/nginx-1.11.3/debian/modules/nchan-1.0.0/src -I src/stream \
                -o objs/addon/redis/rdsstore.o \
                /home/bbigras/nginx/nginx-1.11.3/debian/modules/nchan-1.0.0/src/store/redis/rdsstore.c
/home/bbigras/nginx/nginx-1.11.3/debian/modules/nchan-1.0.0/src/store/redis/rdsstore.c: In function ‘redis_subscriber_callback’:
/home/bbigras/nginx/nginx-1.11.3/debian/modules/nchan-1.0.0/src/store/redis/rdsstore.c:1082:13: error: passing argument 2 of ‘cmp_read_integer’ from incompatible pointer type [-Werror]
             if(cmp_read_integer(&cmp, &msgbuf_size))
             ^
In file included from /home/bbigras/nginx/nginx-1.11.3/debian/modules/nchan-1.0.0/src/store/redis/rdsstore.c:65:0:
/home/bbigras/nginx/nginx-1.11.3/debian/modules/nchan-1.0.0/src/store/redis/cmp.h:237:6: note: expected ‘int64_t *’ but argument is of type ‘intptr_t *’
 bool cmp_read_integer(cmp_ctx_t *ctx, int64_t *d);
      ^
cc1: all warnings being treated as errors
make[3]: *** [objs/addon/redis/rdsstore.o] Error 1
make[3]: Leaving directory `/home/bbigras/nginx/nginx-1.11.3'
make[2]: *** [build] Error 2
make[2]: Leaving directory `/home/bbigras/nginx/nginx-1.11.3'
dh_auto_build: make -j1 returned exit code 2
make[1]: *** [override_dh_auto_build] Error 2
make[1]: Leaving directory `/home/bbigras/nginx/nginx-1.11.3'
make: *** [build] Error 2
```

```
$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/i686-linux-gnu/4.8/lto-wrapper
Target: i686-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 4.8.4-2ubuntu1~14.04.3' --with-bugurl=file:///usr/share/doc/gcc-4.8/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.8 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.8 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-libmudflap --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.8-i386/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.8-i386 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.8-i386 --with-arch-directory=i386 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-targets=all --enable-multiarch --disable-werror --with-arch-32=i686 --with-multilib-list=m32,m64,mx32 --with-tune=generic --enable-checking=release --build=i686-linux-gnu --host=i686-linux-gnu --target=i686-linux-gnu
Thread model: posix
gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.3)
```

**Comment by slact** [comment]
Ah, this is a 32-bit machine warning

**Comment by slact** [comment]
fixed in 3ace08b

**Comment by brunoqc** [comment]
It compiled. :) Thanks!
